node/src/p2p.rs

```diff
@@ -190,65 +330,64 @@ impl Worker {
     #[instrument(level = "trace", skip(self))]
     async fn on_swarm_event(
         &mut self,
-        ev: &SwarmEvent<BehaviourEvent, THandlerErr<Behaviour>>,
+        ev: SwarmEvent<BehaviourEvent, THandlerErr<Behaviour>>,
     ) -> Result<()> {
-        #[allow(clippy::single_match)]
         match ev {
             SwarmEvent::Behaviour(ev) => match ev {
                 BehaviourEvent::Identify(ev) => self.on_identify_event(ev).await?,
-                BehaviourEvent::HeaderEx(ev) => self.on_header_ex_event(ev).await?,
-                BehaviourEvent::KeepAlive(_) => {}
                 BehaviourEvent::Gossipsub(ev) => self.on_gossip_sub_event(ev).await?,
+                BehaviourEvent::HeaderEx(_) | BehaviourEvent::KeepAlive(_) => {}
             },
+            SwarmEvent::ConnectionEstablished { peer_id, .. } => {
+                self.peer_tracker.add(peer_id);
+
+                // This vector is not going to be used later on so deallocate
+                // its internal memory by consuming it with mem::take.
+                for tx in mem::take(&mut self.wait_connected_tx) {
+                    tx.maybe_send(());
+                }
```

Comment:
I'd not use `VecDeque` and `mem::take` for `wait_connected_tx`. You could use the `Vec`, replace `push_back` with `push` and here use the `.drain(..)`

Reply:
I forgot to change it to `Vec`. However even in case of `Vec`, `mem::take` is needed because `.drain(..)` does not shrink the capacity. This is an one-off queue.

Reply:
if it is a one-off queue, then keep this `Vec` as an option. otherwise it's not, if we disconnect from all the peers (network error or whatever) then it'll again behave like so